bit9_connector.py

```diff
@@ -462,11 +520,79 @@ def _upload_file(self, param):
             summary = action_result.update_summary({'upload_status': upload_status})
             try:
                 summary['upload_status_desc'] = self.UPLOAD_STATUS_DESCS[str(upload_status)]
-            except:
+            except Exception:
                 pass
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _list_files(self, param):
+        action_result = self.add_action_result(ActionResult(param))
+        ret_val, limit = self._validate_integer(action_result, param.get('limit', 100), 'Limit', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            'limit': limit
+        }
+
+        endpoint = '/fileUpload'
+
+        ret_val, resp_json = self._make_rest_call(endpoint, action_result, method="get", params=params)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(resp_json)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully fetched available files")
+
+    def _get_file(self, param):
+        action_result = self.add_action_result(ActionResult(param))
+
+        ret_val, file_id = self._validate_integer(action_result, param["file_id"], 'File ID', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            'downloadFile': True
+        }
+
+        endpoint = '/fileUpload/' + str(file_id)
+
+        ret_val, resp_json = self._make_rest_call(endpoint, action_result, method="get")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        filename = resp_json.get('fileName')
+
+        ret_val, resp = self._make_rest_call(endpoint, action_result, method="get", params=params)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        if hasattr(Vault, 'get_vault_tmp_dir'):
+            vault_tmp_dir = Vault.get_vault_tmp_dir()
+        else:
+            vault_tmp_dir = '/opt/phantom/vault/tmp'
+
+        file_loc = vault_tmp_dir + '/' + filename
+        with open(file_loc, 'w') as file:
+            file.write(resp.text)
+
+        vault_ret_dict = phantomrules.vault_add(container=self.get_container_id(),
+                                                file_location=file_loc,
+                                                file_name=filename)
+        if vault_ret_dict[0]:
+            vault_details = {
+                phantom.APP_JSON_VAULT_ID: vault_ret_dict[2],
+                'file_name': filename
+            }
+            action_result.add_data(vault_details)
+            return action_result.set_status(phantom.APP_SUCCESS, "Successfully added file to vault")
+
+        return action_result.set_status(phantom.APP_ERROR, 'Error adding file to vault: {0}'.format(vault_ret_dict))
```

Comment:
In case of failure can we return the message that is being returned by the vault_add API?
It would `message` (preferred) or `vault_ret_dict[1]`.